foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/queries/ReportQueryResult.java

```diff
@@ -277,21 +277,6 @@ protected Object processItem(ReportQuery query, AbstractRecord row, Vector toMan
                 }
             } else {
                 value = row.getValues().get(itemIndex);
-
-                // Verify that the expected result type matches the actual value type
-                if(value != null && query.getSession().getProject().allowResultTypeConversion()) {
```

Comment:
corresponding property should be removed, method deprecated and this particular call removed

Reply:
I added the property so that behavior will only change IF the property is set. The property defaults to turn this fix OFF. It would break existing customers if you update 2.7 to change behavior. 

Customers may have written their applications to expect Integer, even though Boolean is TECHNICALLY correct. Without this property around the fix, customers WILL BREAK and have to change their application implementation.